### PR TITLE
add namespace support by prefixing configured paths

### PR DESCRIPTION
**Pull request by clintongmatthews**

### CODE REVIEW

File: sync.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    return prefixedEnvs
+  }
+}
+
+const buildEnv = ({
+  namespace,
+  prefixes: orderedPrefixes,
+  envCache,
+}) => Object.fromEntries(
+  Object.entries(
```
Comment (clintongmatthews) [inline sync.js:141]: it is unfortunate that the pr looks like this. the index file was renamed to sync and this is the main change. the new index file is just setup to parse varied inputs then delegate to this file. again no changes in here except the joi schema and this function to add the namespace to the env keys

**Review by jrpz — APPROVED** [review]
We'll probably want to update the docs for this, https://github.com/articulate/consul-sync/blob/master/API.md

**Review by kfurness — APPROVED** [review]
Looks good.

image

**Comment by clintongmatthews** [comment]
now that we don't write to process.env I was wondering if we want to expose updates using the stream that is created to the client rather than requiring a function call to get updates. anyway, just thinking out loud

**Comment by clintongmatthews** [comment]
opted to move code to the client app

**Pull request closed without merging**